lib/klaviyo_api/resources/profile.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module KlaviyoAPI
+  class Profile < Base
+    self.prefix += 'v1/'
+
+    self.collection_name = 'people'
+    self.element_name = 'person'
+    self.collection_parser = KlaviyoAPI::Collections::Profile
+
+    class << self
+      def collection_path(prefix_options = {}, query_options = {})
+        super prefix_options, query_options.deep_merge({ api_key: headers['api-key'] })
+      end
+
+      def element_path(id, prefix_options = {}, query_options = {})
+        super id, prefix_options, query_options.deep_merge({ api_key: headers['api-key'] })
+      end
+    end
+
+    def destroy
+      raise KlaviyoAPI::InvalidOperation.new 'Cannot delete Profiles via API.'
+    end
+
+    def create
+      raise KlaviyoAPI::InvalidOperation.new 'Cannot create Profiles via API.'
+    end
+
+    def update
+      run_callbacks :update do
+        # This endpoint does not accept JSON bodies. Additionally, if the API key is in the URL, the rest of the attributes
+        # MUST be in the URL as well. The other option is to have the API key and all attributes in the body as `key=value` pairs.
+        #
+        # To reduce friction, as we already need the API key in the URL for other operations, let's just add all the attributes
+        # to the URL. Unfortunately we will need to include ALL the attributes, not just the changed ones, because it seems
+        # ActiveResource has no support for dirty models. https://github.com/rails/activeresource/issues/308
+        path = self.class.element_path(id, prefix_options, attributes)
+
+        connection.put(path, nil, self.class.headers).tap do |response|
+          load_attributes_from_response(response)
+        end
+      end
+    end
+
+    private
+
+    def method_missing(method_symbol, *arguments)
```

Comment:
I like this but aren't there cases where there is a `$first_name` and `first_name` in the payload? Now I'd like to assume that they are both the same thing but well, yeah.

Reply:
Yup - hence [my comment](https://github.com/rewindio/klaviyo_api/pull/3/files#diff-d4aea8ecf9da888bceafa8edf7585f94R54) and [my test ](https://github.com/rewindio/klaviyo_api/pull/3/files#diff-0009d0f9d036c9df569f02d0d91f9d0dR100).

If I do nothing, then this object always has to be accessed using `profile.attributes['title']`.

If I check for `title` first and fallback to `$title`, then you can't add custom attributes like `profile.my_attr = 'hi'` because it will complain that `$my_attr` is invalid or something (I forgot what I saw yesterday, exactly).

If I check for `$title` first and fallback to `title`, then, yes, you lose method access to `title` in favour of `$title`, but at least you can add custom attributes because `profile.my_attr` will add the attribute `my_attr`, and Klaviyo needs custom attributes to NOT start with $. So that made the most sense to me.

There is not real winning here. Only lesser losing.